### PR TITLE
New Adapter: Clickonometrics

### PR DESCRIPTION
**Pull request by mjaworskiccx**

### CODE REVIEW

File: openrtb_ext/imp_ccx.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package openrtb_ext
+
+type ExtImpCcx struct {
+	PlacementID string `json:"pid"`
```
Comment (AlexBVolcy) [inline openrtb_ext/imp_ccx.go:4]: Could you rename the json path to be `json:"placementId"`

File: adapters/ccx/ccx.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
```
Comment (AlexBVolcy) [inline adapters/ccx/ccx.go:62]: Is it possible to cover these two error cases?

**Review by AlexBVolcy — COMMENTED** [review]
Please link your docs PR

File: openrtb_ext/imp_ccx.go (hunk 1 above)
Comment (mjaworskiccx) [reply openrtb_ext/imp_ccx.go:4]: https://github.com/prebid/prebid.github.io/pull/4065

File: adapters/ccx/ccx.go (hunk 2 above)
Comment (mjaworskiccx) [reply adapters/ccx/ccx.go:62]: Done

File: openrtb_ext/imp_ccx.go (hunk 1 above)
Comment (mjaworskiccx) [reply openrtb_ext/imp_ccx.go:4]: Is that necessary? 
My placement param is pid in static\bidder-params\ccx.json
Do i not understand something?

Comment (mjaworskiccx) [reply openrtb_ext/imp_ccx.go]: Done

**Comment by AlexBVolcy** [comment]
There is a PR https://github.com/prebid/prebid-server/pull/2370 that is being merged today that will cause conflicts to new adapter PRs. 

To resolve these conflicts, you'll need to do the following:

In `ccx.go`, update the `Builder()` function signature to include a new parameter `server config.Server`

In `ccx_test.go`, update any test that includes a call to `Builder()` to pass a `config.Server` object, to that call (i.e. `config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "datacenter"}`)

After the merge, you can look at other adapters to see how the server object is implemented as an example.

File: adapters/ccx/ccx.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}}
+	}
```
Comment (SyntaxNode) [inline adapters/ccx/ccx.go:32]: You do not need this check. The core logic of Prebid Server will only call this method if there is at least one impression.

**Review by hhhjort — APPROVED** [review]
LGTM

File: static/bidder-info/ccx.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+endpoint: "https://delivery.clickonometrics.pl/ortb/prebid/bid"
+maintainer:
+  email: "it@clickonometrics.pl"
+gvlVendorID: 773
+capabilities:
+  site:
+    mediaTypes:
+      - banner
```
Comment (AlexBVolcy) [inline static/bidder-info/ccx.yaml:8]: I'm noticing a discrepancy between the supported media types in this file vs. what's present in your dev docs. Here it says banner is the media type supported, but the doc says video. Could you address this?

**Review by AlexBVolcy — COMMENTED** [review]
Looks good, one docs related comment

File: static/bidder-info/ccx.yaml (hunk 4 above)
Comment (mjaworskiccx) [reply static/bidder-info/ccx.yaml:8]: added

**Comment by bsardo** [comment]
@mjaworskiccx can you please merge with master when you get a chance? The 1.19 validation checks are failing because we recently upgraded to 1.19 and that version has made some updates to `go fmt` that affect the formatting in some of the files you modified. Thanks!

**Comment by mjaworskiccx** [comment]
done

**Comment by bsardo** [comment]
Hi @mjaworskiccx, it looks like you have some failing tests due to some recent changes we made on master. You'll need to update the reference to the mxmcherry library with the prebid version of the ortb library.

**Comment by mjaworskiccx** [comment]
done

**Review by hhhjort — APPROVED** [review]

File: static/bidder-info/ccx.yaml (hunk 4 above)
Comment (AlexBVolcy) [reply static/bidder-info/ccx.yaml:8]: Thanks for updating your .yaml file, but could you also update your docs PR to include the banner media type as well?

Comment (bsardo) [reply static/bidder-info/ccx.yaml:8]: @AlexBVolcy I see [PR #4127](https://github.com/prebid/prebid.github.io/pull/4127) was opened to address this.

**Review by AlexBVolcy — APPROVED** [review]

**Pull request merged**